src/modules/kubernetes.rs

```diff
@@ -148,10 +176,19 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         .map(|filename| get_kube_ctx_component(filename, kube_ctx.clone()))
```

Comment:
Something I just realized while poking around this module: it looks like in all subsequent uses of the `ctx_components` variable, we iterate over the components, check if they're `Some`, and then check if one of their components is a `Some`.

Why don't we just filter the vector here to only contain non-`None` components?

Reply:
you mean add a `.filter_map(|x| x)` here to filter out the `None` and declare it as `Vec<KubeCtxComponents>`?

Reply:
On further reflection, I think we can change this `map` to `filter_map` and get the same result.